tests/flytekit/common/workflows/sagemaker.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import random
+from operator import add
+
+from six.moves import range
+
+from flytekit.sdk.tasks.sagemaker.trainingjob import SdkSimpleTrainingJobTask, DataChannel
+from flytekit.sdk.tasks.sagemaker.hpojob import SdkSimpleHPOJobTask, HPOJobTuningStrategy, HPOJobObjective
+from flytekit.sdk.tasks.sagemaker.parameter_range import IntegerParameterRange, ContinuousParameterRange, CategoricalParameterRange
+from flytekit.sdk.tasks.sagemaker import trainingjob_task
+from flytekit.sdk.tasks import inputs, outputs, python_task
+from flytekit.sdk.types import Types
+from flytekit.sdk.workflow import workflow_class, Input, Output
+import xgboost as xgb
+
+example_hyperparams = {
+    "base_score": "0.5",
+    "booster": "gbtree",
+    "csv_weights": "0",
+    "dsplit": "row",
+    "grow_policy": "depthwise",
+    "lambda_bias": "0.0",
+    "max_bin": "256",
+    "max_leaves": "0",
+    "normalize_type": "tree",
+    "objective": "reg:linear",
+    "one_drop": "0",
+    "prob_buffer_row": "1.0",
+    "process_type": "default",
+    "rate_drop": "0.0",
+    "refresh_leaf": "1",
+    "sample_type": "uniform",
+    "scale_pos_weight": "1.0",
+    "silent": "0",
+    "sketch_eps": "0.03",
+    "skip_drop": "0.0",
+    "tree_method": "auto",
+    "tweedie_variance_power": "1.5",
+    "updater": "grow_colmaker,prune",
+}
+
+simple_xgboost_trainingjob_task = SdkSimpleTrainingJobTask(
+    # The problem is that SageMaker currently doesn't support fractional resources
+    # (that is, you have to be billed for the entire instance even though your program
+    # uses only a part of it).
+    #
+    # This aforementioned billing policy of SageMaker conflicts with the current semantic
+    # in flytekit around resources. So If we allow users to specify cpu_limit and memory_
+    # limit for SageMaker tasks, users might be misled and found that they are "overcharged"
+    # for the resource they didn't claim.
+    #
+    # Exposing InstanceType and InstanceCount to the users seems to be the most viable workaround
+    # I can think of at the moment to address the above issue.
+    trainingjob_conf={
+        "InstanceType": "ml.m4.xlarge",
+        "InstanceCount": 1,
+        "VolumeSizeInGB": 25,
+    },
+    stopping_condition={
+        "MaxRuntimeInSeconds": 43200,
+        "MaxWaitTimeInSeconds": 43200
+    },
+    algorithm_specification={
+        "TrainingInputMode": "File",
+        "AlgorithmName": "xgboost",
```

Comment:
samething, an enumeration and for algorithm besides the one supported directly (byo) just have "OTHERS"/"BYO" as an option